presto-native-execution/presto_cpp/main/tests/LinuxMemoryCheckerTest.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/LinuxMemoryChecker.cpp"
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+#include "velox/common/base/VeloxException.h"
+#include "velox/common/base/tests/GTestUtils.h"
+
+namespace fs = boost::filesystem;
+
+namespace {
+std::string getStatsFilePath(const std::string& fileName) {
+  std::string currentPath = fs::current_path().c_str();
+  if (boost::algorithm::ends_with(currentPath, "fbcode")) {
+    return currentPath +
+        "/github/presto-trunk/presto-native-execution/presto_cpp/main/tests/examples/" +
+        fileName;
+  }
+
+  // CLion runs the tests from cmake-build-release/ or cmake-build-debug/
+  // directory. Hard-coded json files are not copied there and test fails with
+  // file not found. Fixing the path so that we can trigger these tests from
+  // CLion.
+  boost::algorithm::replace_all(currentPath, "cmake-build-release/", "");
+  boost::algorithm::replace_all(currentPath, "cmake-build-debug/", "");
+
+  return currentPath + "/presto_cpp/main/tests/examples/" + fileName;
+}
+} // namespace
+
+namespace facebook::presto {
+class LinuxMemoryCheckerTest : public testing::Test {};
+
+TEST_F(LinuxMemoryCheckerTest, memoryStatFileV1) {
+  // Default config.
+  LinuxMemoryChecker memChecker(PeriodicMemoryChecker::Config{});
+
+  // Testing cgroup v1/v2 memory.stat file.
+  auto statFilePath = getStatsFilePath("memoryStatFileV1.stat");
```

Comment:
Use a common function for these three lines as it is the same for all tests. Make LinuxMemoryChecker  and this function part of the LinuxMemoryCheckerTest class.